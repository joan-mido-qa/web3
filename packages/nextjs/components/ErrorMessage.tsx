interface Props {
  message: string;
}

export default function ErrorMessage({ message }: Props) {
  return (
    <div className="z-50 absolute bottom-5 left-5 bg-red-100 border-l-4 border-red-500 text-red-700 p-4" role="alert">
      {message}
    </div>
  );
}
