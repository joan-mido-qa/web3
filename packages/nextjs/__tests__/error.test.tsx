import ErrorMessage from "@/components/ErrorMessage";
import { render, screen } from "@testing-library/react";

describe("Error Message", () => {
  test("Should render ErrorMessage component unchanged", async () => {
    const { container } = render(<ErrorMessage message="An Error" />);
    expect(container).toMatchSnapshot();
    expect(screen.getByRole("alert").innerText).toBe("An Error");
  });
});
