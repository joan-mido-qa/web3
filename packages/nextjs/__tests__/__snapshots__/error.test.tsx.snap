// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Message Should render ErrorMessage component unchanged 1`] = `
<div>
  <div
    class="z-50 absolute bottom-5 left-5 bg-red-100 border-l-4 border-red-500 text-red-700 p-4"
    role="alert"
  >
    An Error
  </div>
</div>
`;
